### PR TITLE
Fix some e2e tests in custom-column and view-data

### PR DESCRIPTION
**Pull request by rafpaf**
This PR fixes some e2e tests that broke when #49442 merged

Running stress tests now. Will paste results as they come in:

### should allow to use `if` function

https://github.com/metabase/metabase/actions/runs/12394723764
image

### should allow you to reorder and hide columns in the table header
https://github.com/metabase/metabase/actions/runs/12394798005
image

### allows editing sandboxed access in the group focused view
https://github.com/metabase/metabase/actions/runs/12394835744
image

### CODE REVIEW

**Comment by rafpaf** [comment]
* **#51431** <a href="https://app.graphite.dev/github/pr/metabase/metabase/51431?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/51431?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: e2e/test/scenarios/permissions/view-data.cy.spec.js [hunk 1]
```diff
@@ -334,17 +334,10 @@ H.describeEE(
 
         cy.findByText("Revoke access").click();
       });
-      // Page leave confirmation should be on top
-      cy.findByTestId("leave-confirmation")
-        .as("leaveConfirmation")
-        .findByText("Discard your changes?")
-        .should("be.visible");
```
Comment (rafpaf) [inline e2e/test/scenarios/permissions/view-data.cy.spec.js:341]: I added this but it's apparently not needed here, and breaks the test

File: e2e/support/helpers/e2e-custom-column-helpers.ts [hunk 2]
```diff
@@ -18,16 +18,12 @@ export function enterCustomColumnDetails({
...
-    .focus()
-    .clear()
-    .type(formula);
+  const input = cy.get(".ace_text-input").first().as("formula");
+
+  input.should("exist").focus().clear().type(formula);
 
   if (blur) {
-    cy.get("@formula").blur();
+    input.blur();
```
Comment (filiphric) [inline e2e/support/helpers/e2e-custom-column-helpers.ts:26]: this approach seems a bit off for me. once there’s an alias, there’s no point of using the `input` variable. I think the old approach was better when it comes to Cypress syntax. On runtime I believe these do pretty much the same thing

File: e2e/test/scenarios/custom-column/custom-column.cy.spec.js [hunk 3]
```diff
@@ -799,7 +799,7 @@ describe("scenarios > question > custom column", () => {
           formula: 'notIn("Gadget", [Vendor], [Category])',
           name: "InColumn",
         });
-        cy.button("Update").click();
+        cy.button("Update").should("not.be.disabled").click();
```
Comment (filiphric) [inline e2e/test/scenarios/custom-column/custom-column.cy.spec.js:802]: Cypress has [actionability checks](https://docs.cypress.io/app/core-concepts/interacting-with-elements#Actionability) before interacting with elements. This `.should("not.be.disabled")` assertion doesn’t do anything extra

**Review by filiphric — APPROVED** [review]

**Pull request merged**